sklearn/utils/tests/test_array_api.py

```diff
@@ -112,6 +113,21 @@ def test_array_api_wrapper_astype():
     assert X_converted.dtype == xp.float32
 
 
+def test_array_api_wrapper_maximum():
+    """Test _ArrayAPIWrapper for ArrayAPIs that is not NumPy."""
+    array_api_strict = pytest.importorskip("array_api_strict")
+    xp_ = _AdjustableNameAPITestWrapper(array_api_strict, "array_api_strict")
```

Comment:
Why would `array_api_strict` need such a wrapper for `maximum` in the first place?

According to https://github.com/data-apis/array-api-strict/blob/7c55e4712b706128366a12fc3bea82d182b9f1a6/array_api_strict/_elementwise_functions.py#L658 it already supports it in version 2.


```python
>>> import array_api_strict as xp
>>> xp.__version__
'2.0.1'
>>> xp.set_array_api_strict_flags(api_version="2023.12")
<ipython-input-4-660897219bbc>:1: UserWarning: The 2023.12 version of the array API specification is still preliminary. Some functions are not yet implemented, and it has not been fully tested.
  xp.set_array_api_strict_flags(api_version="2023.12")
>>> xp.maximum(xp.arange(5), xp.full(shape=5, fill_value=3))
Array([3, 3, 3, 3, 4], dtype=array_api_strict.int64)
```

and we automatically set `api_version="2023.12"` in our `get_namespace`.

Reply:
`array_api_strict` does not need this. This test has been added because codecov wanted these lines covered and `cupy.array_api` is not available everywhere. Hence to test something within the `_ArrayAPIWrapper` we have to pick another api, `array_api_strict` in this case.

Reply:
Please add an inline comment to explain that this is to test a wrapper that would be otherwise only be needed for cupy which is not available on our coverage-enabled PR CI.